apps/files_sharing/lib/External/Manager.php

```diff
@@ -538,4 +540,84 @@ private function getShares($accepted) {
 
 		return $result ? $shares->fetchAll() : [];
 	}
+
+	/**
+	 * Test whether the specified remote is accessible
+	 */
+	protected function testUrl(IClientService $clientService, string $remote, bool $checkVersion = false, bool $throwConnectException = false): bool {
+		try {
+			$client = $clientService->newClient();
+			$response = \json_decode($client->get(
+				$remote,
+				[
+					'timeout' => 3,
+					'connect_timeout' => 3,
+				]
+			)->getBody());
+
+			if ($checkVersion) {
+				return !empty($response->version) && \version_compare($response->version, '7.0.0', '>=');
+			}
+
+			return \is_object($response);
+		} catch (ConnectException $e) {
+			if ($throwConnectException) {
+				throw $e;
+			}
+		} catch (\Exception $e) {
+		}
+		return false;
+	}
+
+	public function testRemoteUrl(IClientService $clientService, string $remote) {
+		$parsed_url = parse_url($remote, PHP_URL_HOST);
+		if (\is_string($parsed_url)) {
+			$remote = $parsed_url;
+		} else {
+			$parsed_url = parse_url('http://' . $remote, PHP_URL_HOST);
+			if (\is_string($parsed_url)) {
+				$remote = $parsed_url;
+			}
+		}
+		try {
+			// one initial connectivity check on https and http
+			if (!$this->checkConnectivity($clientService, $remote)) {
+				return false;
+			}
+			if ($this->testUrl($clientService, 'https://' . $remote . '/ocs-provider/') ||
+				$this->testUrl($clientService, 'https://' . $remote . '/ocs-provider/index.php') ||
+				$this->testUrl($clientService, 'https://' . $remote . '/status.php', true)
+			) {
+				return 'https';
+			}
+
+			if (
+				$this->testUrl($clientService, 'http://' . $remote . '/ocs-provider/') ||
+				$this->testUrl($clientService, 'http://' . $remote . '/ocs-provider/index.php') ||
+				$this->testUrl($clientService, 'http://' . $remote . '/status.php', true)
+			) {
+				return 'http';
+			}
+		} catch (ConnectException $e) {
+			\OC::$server->getLogger()->logException($e, ['message' => "Remote $remote not reachable"]);
+		}
+
+		return false;
+	}
+
+	private function checkConnectivity(IClientService $clientService, string $remote): bool {
+		foreach (['http://', 'https://'] as $schema) {
+			try {
+				if ($this->testUrl($clientService, $schema . $remote . '/status.php', false, true)) {
+					\OC::$server->getLogger()->error("Remote $schema$remote/status.php reachable");
+					return true;
+				}
```

Comment:
```suggestion
				if ($this->testUrl($clientService, $schema . $remote . '/status.php', false, true)) {
					return true;
				}
				\OC::$server->getLogger()->error("Remote $schema$remote/status.php not reachable");
```

If the host:port of the remote is completely down, then `testUrl` is returning false, but I get nothing in the log about it.
Maybe move the error logging to the unhappy path, and don't log when reachable.

Reply:
thx for catching this :pray: